### ISSUE TITLE
apiserver port changed when docker restart kind container

### ISSUE THREAD

**Issue by win5do** [kubernetes-sigs/kind#1309]
kind version:
```text
kind v0.7.0 go1.13.6 darwin/amd64
```

docker version:
```text
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea
 Built:             Wed Nov 13 07:22:34 2019
 OS/Arch:           darwin/amd64
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea
  Built:            Wed Nov 13 07:29:19 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```
<img width="1134" alt="WechatIMG30" src="https://user-images.githubusercontent.com/22112659/73712799-36bf2200-4746-11ea-858b-e3a69e749299.png">

I used  to use kind@0.5.1, it is OK. 
Sometimes I want to reboot and get back to my previous work.

**Comment by win5do** [comment]
Now, I have to create cluster with a config file to pinned apiserver port, like this:
```yaml
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
networking:
  apiServerAddress: "127.0.0.1"
  apiServerPort: 22000
```

**Comment by BenTheElder** [comment]
We ask docker to get a random port to work with remote docker.

Restart support has an existing tracking issue, we're actively looking into it.

**Comment by BenTheElder** [comment]
https://github.com/kubernetes-sigs/kind/issues/148

**Comment by BenTheElder** [comment]
going to tweak this behavior, regardless of restart support (which we are intending to land anyhow) this is sub-optimal.
/assign

**Comment by BenTheElder** [comment]
https://github.com/kubernetes-sigs/kind/pull/1353 will resolve this

**Comment by BenTheElder** [comment]
resolved in #1353
